src/test/java/com/sigma/cinematicketpro/TestUtils.java

```diff
@@ -147,4 +154,27 @@ public static RegistrationRequest getValidRegistrationRequest() {
                 .password("secure_pass")
                 .build();
     }
+
+    public static List<TMDBGenre> getTMDBGenresList() throws IOException {
+        return mapJsonToObject("tmdb/genres/genres.json", new TypeReference<>() {});
+    }
+
+    public static Map<Integer, TMDBGenre> getTMDBGenresIdGenreMap() throws IOException {
+        return getTMDBGenresList().stream()
+                .collect(toMap(TMDBGenre::getId, identity()));
+    }
+
+    public static List<TMDBMovie> getTMDBMoviesWithMappedGenresTitles() throws IOException {
+        return mapJsonToObject("tmdb/movies/trending-movies-mapped-genres-titles.json", new TypeReference<>() {});
+    }
+
+    public static List<TMDBMovie> getTMDBMoviesWithoutGenresTitles() throws IOException {
+        return mapJsonToObject("tmdb/movies/trending-movies-no-genres-titles.json", new TypeReference<>() {
+        });
+    }
+
+    public static <T> T mapJsonToObject(String resourcePath, TypeReference<T> typeReference) throws IOException {
```

Comment:
I like generics but in general they are objects - if you have a chance to use strict types - go for it

Reply:
discussed